src/domain/exceptions.py

```
@@ -0,0 +1,49 @@
+class AirQualityException(Exception):
+    def __init__(self, message: str, details: Optional[Dict[str, Any]] = None):
+        super().__init__(message)
+        self.details = details or {}
+
+
+class DataSourceException(AirQualityException):
+    pass
+
+
+class RateLimitException(DataSourceException):
+    def __init__(self, message: str, retry_after: Optional[int] = None, **kwargs):
```

Ruff (0.12.2)

Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)